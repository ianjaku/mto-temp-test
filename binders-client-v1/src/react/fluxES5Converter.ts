/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
