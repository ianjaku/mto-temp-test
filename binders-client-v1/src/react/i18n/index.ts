import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
