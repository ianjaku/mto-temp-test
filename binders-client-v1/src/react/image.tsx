/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
