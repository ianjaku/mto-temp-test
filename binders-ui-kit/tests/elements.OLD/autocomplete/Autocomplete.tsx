import * as React from "react";
import Autocomplete, { IAutocompleteItem } from "../../../src/elements/autocomplete";
import { create } from "react-test-renderer";
import wrapWithTheme from "../../themeHelper";

const data: IAutocompleteItem[] = [
    {
        id: "0",
        label: "joana darek",
        rawLabel: "joana darek",
        value: "joana@manual.to",
    },
    {
        id: "1",
        label: "ann",
        rawLabel: "ann",
        value: "anna@gmail.com",
    },
    {
        id: "2",
        label: "darek chec",
        rawLabel: "darek chec",
        value: "dariusz.chec.ma.dlugiego.maila@gmail.com",
    },
    {
        id: "3",
        label: "darek chec2",
        rawLabel: "darek chec2",
        value: "dariusz.chec3.ma.dlugiego.maila@gmail.com",
    },
    {
        id: "4",
        label: "darek chec23",
        rawLabel: "darek chec23",
        value: "dariusz.chec2.ma.dlugiego.maila@gmail.com",
    },
    {
        id: "5",
        label: "darek chec4",
        rawLabel: "darek chec4",
        value: "dariusz.chec4.ma.dlugiego.maila@gmail.com",
    },
    {
        id: "6",
        label: "darek chec5",
        rawLabel: "darek chec5",
        value: "dariusz.chec5.ma.dlugiego.maila@gmail.com",
    },
    {
        id: "7",
        label: "mam bardzo dlugie imie darek",
        rawLabel: "mam bardzo dlugie imie darek",
        value: "krotki@mail",
    },
];

let exampleValue = 100;
const onClick = () => exampleValue++;
const onClick2 = () => exampleValue--;


function createComponent() {
    return wrapWithTheme(<Autocomplete selectedItems={[]} onDeleteChipClick={onClick2} data={data} onAddNewClick={onClick} />);
}

describe("Button", () => {

    beforeEach(() => {
        exampleValue = 100;
    });


    test("Base Autocomplete (snapshot)", () => {
        const modal = create(createComponent());
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    // todo: tests

    // test("Autocomplete prompts (mount: enzyme+jsdom)", () => {
    //     const elementText = mount(createComponent());
    //     elementText.setState({searchTerm: "darek"});
    //     const promptsLen = elementText.find("li").length;

    //     expect(promptsLen).toEqual(7);
    // });

    // test("Autocomplete onClick (mount: enzyme+jsdom)", () => {
    //     const elementText = mount(createComponent());
    //     elementText.setState({searchTerm: "darek"});
    //     const prompts = elementText.find("li").at(0);
    //     prompts.simulate("click");

    //     expect(exampleValue).toEqual(101);
    // });
});
