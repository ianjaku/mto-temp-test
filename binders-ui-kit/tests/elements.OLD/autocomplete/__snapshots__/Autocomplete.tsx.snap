// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Base Autocomplete (snapshot) 1`] = `
<div>
  <div
    className="autocomplete"
  >
    <textarea
      className="autocomplete-textarea"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Invite new user or user group"
      rows={1}
      value=""
    />
  </div>
</div>
`;
